src/Features/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs

```diff
@@ -120,6 +120,30 @@ public void StopTracking(Uri uri)
         _lspMiscellaneousFilesWorkspace?.TryRemoveMiscellaneousDocument(uri);
 
         LspTextChanged?.Invoke(this, EventArgs.Empty);
+
+        ReloadContentsFromDiskInMutatingWorkspace(uri);
+
+        return;
+
+        void ReloadContentsFromDiskInMutatingWorkspace(Uri uri)
+        {
+            var registeredWorkspaces = _lspWorkspaceRegistrationService.GetAllRegistrations();
+            foreach (var workspace in registeredWorkspaces)
+            {
+                if (workspace is not IMutatingLspWorkspace mutatingWorkspace)
+                    continue;
+
+                foreach (var document in workspace.CurrentSolution.GetDocuments(uri))
+                {
+                    if (document.FilePath is null)
+                        continue;
+
+                    mutatingWorkspace.UpdateTextIfPresent(
+                        document.Id,
+                        new WorkspaceFileTextLoader(workspace.Services.SolutionServices, document.FilePath, defaultEncoding: null));
```

Comment:
@jasonmalinowski if there is another sort of loader we should prefer here.  Effectively, when the lsp side of things says that the file is clsoed, we want to tell the workspace "well, you can go back to reading it from disk".  I'm not sure if there's a standard/better way of doing that versus explicitly just creating this sort of loader *directly* here.

Reply:
Rather than this creating the text loader, I'd imagine the method should just be UpdateTextToUseLoaderAgain() or something like that, and let the workspace decide what the appropriate loader type is.

Reply:
Is there a helper already in the workspace to do that?  I'm trying to figure out how we would determine that for all workspaces.   I'll ping you next week about this.